### DISCUSSION TITLE
SNOPT not working with the docker (topology optimization tutorial)

### DISCUSSION THREAD

**Discussion by AGHSMI** [mdolab/dafoam#94] [Tutorials] [answered]
Hello,

I am trying to solve the topology optimization tutorial, and have copied all .f files (except snopth.c) in snopt7/src to /home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pySNOPT/source, and ran "python setup.py" with no error. However, it seems SNOPT is not linked to DAFoam, because I receive this mesage:

_Traceback (most recent call last):
  File "runScript.py", line 184, in <module>
    opt = OPT(args.opt, options=optOptions)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pyOpt_optimizer.py", line 941, in OPT
    return opt(*args, **kwargs)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pySNOPT/pySNOPT.py", line 227, in __init__
    raise Error("There was an error importing the compiled snopt module")
pyoptsparse.pyOpt_error.Error_

 Could anyone please help me out. 
P.S.: One other issue I noticed is that my SNOPT version does not have f2py/ subfolder. Is it also important?

Thanks and regards,

**Comment by friedenhe** [comment, marked as answer]
Follow the exact instructions from [here](https://dafoam.github.io/mydoc_installation_source.html#compile-snopt-and-ipopt-for-pyoptsparse-optional) to compile SNOPT. You need to start and log in to a DAFoam docker container, download pyoptsparse-2.3.0 (you can put it anywhere), copy SNOPT source files to pyoptsparse-2.3.0/pyoptsparse/pySNOPT/source, and run `pip install .` This will compile snopt.so and save it to $HOME/dafoam/packages/miniconda3/lib/python3.6/site-packages/pyoptsparse/pySNOPT in the container. You will have to do this every time you log in to the docker container. This is because everything will be deleted after you exit the docker container. So I suggest you compile the dafoam package from scratch for this case.

**Reply by AGHSMI** [reply]
Thanks for the quick reply. I will try to proceed. If it didn't work, I will build DAFoam from the source.

**Reply by AGHSMI** [reply]
Your answer helped me out. The problem was that the pyoptsparse had to be reinstalled.
